apollo-federation/src/sources/connect/json_selection/parser.rs

```diff
@@ -280,11 +368,43 @@ impl PathSelection {
             PathSelection::Empty => None,
         }
     }
+
+    /// Find the next subselection, traversing nested chains if needed. Returns a mutable reference
+    pub(crate) fn next_mut_subselection(&mut self) -> Option<&mut SubSelection> {
+        match self {
+            PathSelection::Var(_, path) => path.next_mut_subselection(),
+            PathSelection::Key(_, path) => path.next_mut_subselection(),
+            PathSelection::Selection(sub) => Some(sub),
+            PathSelection::Empty => None,
+        }
+    }
+
+    pub fn pretty_print(&self, indentation: Option<usize>) -> Result<String, std::fmt::Error> {
+        let mut result = String::new();
+
+        match self {
+            PathSelection::Var(var, path) => {
+                let rest = path.pretty_print(indentation)?;
+                write!(result, "${var}{rest}")?;
```

Comment:
The `var` name here already includes the `$`, and may sometimes be simply `$`. I can fix this, but I have to say: if you wrote unit tests for this code, you would have noticed this.

Reply:
Fixed in https://github.com/apollographql/router/pull/5248